test/common/testlib.py

```diff
@@ -1131,6 +1155,22 @@ def skipPackage(*args):
     return lambda func: func
 
 
+def nondestructive(testEntity):
+    """Tests decorated as nondestructive will all run against the same VM.
+    Can be used on test classes and individual test methods.
+    """
+    def is_test_function(member):
+        return inspect.isfunction(member) and member.__name__.startswith("test")
+    if inspect.isclass(testEntity):
```

Comment:
sorry i didn't suggest this earlier: why not use `issubclass()` here to make sure that it's a `MachineCase`?

Reply:
testEntity might not be a class, but it could be an extra check?